security.md

```diff
@@ -1,60 +1,172 @@
+**Ensure you set the right `Cache-Control` for CDN to not cache your 
+response** Using `Cache-Control: no-cache` is NOT sufficient, use 
+**`Cache-Control: private`** for this purpose. See 
+[Cache-control semantics](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) 
+and 
+[Fastly config](https://developer.fastly.com/learning/concepts/cache-freshness/#preventing-content-from-being-cached)
 
+## Secrets
 **Keep secrets out of repositories.** Application secrets must not be
 stored in version control, even if the repo is "private". To prevent
 accidental leakage, files containing secrets should be stored away
-from the project, completely outside of the source tree.
+from the project, completely outside the source tree.
+
+**Secrets should be rotated regularly.** Automate the rotation if
+possible.
+
+**Some non-secrets should still be kept out of the public domain.** 
+"Private" information that does not constitute a secret but may make
+an exploit easier to achieve should be kept in private repositories.
+
+Also see the guidance on 
+[github repository contents](https://github.com/guardian/recommendations/blob/main/github.md#repository-contents)
+
+
 
 ## AWS
 
+Centralised Reporting from AWS services and guidance is available in
+[Security HQ](https://security-hq.gutools.co.uk).
+
+### Security patches for Compute services
+See the [patching guidance below](#patching)
+
+### Credentials
+
 **Do not use permanent AWS credentials.** Applications running in AWS
 should use `assumeRole` to gain access to other AWS resources. For
 local development you should use AWS profiles to manage credentials
 and these should be temporary credentials (e.g. provided by Janus).
 
+**Permanent credentials should be rotated every 365 days.** (When they
+cannot be replaced by temporary credentials.)
+
 **Ensure that at least 2 people have IAM logins for each AWS account.**  
 In the event of a VPN or Janus outage this will ensure that teams can
-still access their account. IAM users should have 2FA enabled and no
+still access their account. These IAM users should have 2FA enabled and no
 active access keys.
 
-**Use Security HQ to get an overview of our AWS accounts.**
-Centralised Reporting from AWS services, including Trusted Advisor, is
-available in [Security HQ](https://security-hq.gutools.co.uk) and can be
-used to monitor the security status of our AWS accounts.
+### SSH Access
+
+**Direct access to production infrastructure should be a last resort.** 
+Prefer automating common actions and shipping telemetry data extrernally,
+which should reduce the need.
 
 **Use SSM-Scala for SSH access.**
 [SSM-scala](https://github.com/guardian/ssm-scala#ssm-scala) is a tool
 executing commands on EC2 instances authenticated by IAM credentials.
 The SSM Agent is baked into many of our AMI images and should be
 preferentially used to permanent keys.
 
+### Security Groups
+
+**Security Groups egress/ingress rules should be locked down as much as
```

Comment:
Should we mention running EC2 instances in the private subnet of the VPC and only the load balancer in the public subnet?

Reply:
Hmm, I don't think we have any "official" docs. We should create them, though out of scope from this PR.

Reply:
👍 have updated now